Дипломный проект. Этап 3. Дополнительная функциональность.

ViktorSalk opened the pull request:
Реализация — комментариев к событиям.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.
5) Обновил постман-тесты feature.
6) Обновил функциональность comment.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность: комментарии к событиям;.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.
5) Обновил постман-тесты feature.
6) Обновил функциональность comment.
7) Добавил ссылку на пул-реквест.

afomkina commented on README.md:
```diff
@@ -1,2 +1,4 @@
 # java-explore-with-me
```
Распиши, пожалуйста, README более подробно:
* описание проекта
* основные endpoint'ы
* описание доп фичи

ViktorSalk replied:
Добавил описание в README.

afomkina commented on ewm-main-service/src/main/java/ru/practicum/ewm/service/impl/EventServiceImpl.java:
```diff
@@ -369,9 +371,17 @@ public List<EventShortDto> getAllEventFromPublic(SearchEventParams searchEventPa
                 .stream().map(EventMapper::toEventShortDto).collect(Collectors.toList());
         Map<Long, Long> viewStatsMap = getViewsAllEvents(resultEvents);
 
+        List<CountCommentsByEventDto> commentsCountMap = commentRepository.countCommentByEvent(
```
Выражения со стримами, особенно такие длинные, лучше разбивать по строкам, чтобы улучшить читаемость.
Поправь для всех стримов в данном классе

ViktorSalk replied:
Спасибо) В EventServiceImpl все длинные стримы разбиты на несколько строк, улучшена структура и добавлены разделения, для улучшения читаемости.

afomkina commented on ewm-main-service/src/main/java/ru/practicum/ewm/dto/comment/CommentDto.java:
```diff
@@ -0,0 +1,26 @@
+    private Long id;
+    private String text;
+    private Long authorId;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
```
Паттерн `yyyy-MM-dd HH:mm:ss` используется несколько раз в проекте. Чтобы избежать дублирования и случайных ошибок, лучше вынести его в публичную константу

ViktorSalk replied:
Вывел паттерн даты в отдельную константу и обновил классы.

afomkina reviewed (changes requested):
Получилась отличная фича, код выглядит чисто и структурировано 👍

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность: комментарии к событиям;.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.
5) Обновил постман-тесты feature.
6) Обновил функциональность comment.
7) Добавил ссылку на пул-реквест.
8) Вывел паттерн даты в отдельную константу.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность: комментарии к событиям;.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.
5) Обновил постман-тесты feature.
6) Обновил функциональность comment.
7) Добавил ссылку на пул-реквест.
8) Вывел паттерн даты в отдельную константу.
9) В EventServiceImpl все длинные стримы разбиты на несколько строк, улучшена структура и добавлены разделения, для улучшения читаемости.

Viktor3642 committed:
Дипломный проект. Этап 3. Дополнительная функциональность: комментарии к событиям;.
1) Создал ветку feature_comments.
2) Добавил функциональность comment.
3) Добавил постман feature.
4) Добавил Сервис CommentServiceImpl.
5) Обновил постман-тесты feature.
6) Обновил функциональность comment.
7) Добавил ссылку на пул-реквест.
8) Вывел паттерн даты в отдельную константу.
9) В EventServiceImpl все длинные стримы разбиты на несколько строк, улучшена структура и добавлены разделения, для улучшения читаемости.
10) Добавил Описание в README.
